### DISCUSSION TITLE
Problem with rendering a bitmap as a texture

### DISCUSSION THREAD

**Discussion by inveders** [google/filament#4393] [Q&A] [answered]
Hi ! A set a bitmap as a texture on a rectangle (composed by two triangle), all is fine with the .mat file, but one triangle has a really good texture and the other triangle render the texture as is blur.

Is it an issue with the size of my bitmap (I forced it to w:100 and h:100)?

**Comment by bejado** [comment, marked as answer]
Do you have a screenshot? I suspect the UV coordinates of one of the triangles are incorrect.

**Reply by inveders** [reply]
Hi @bejado I'll try to show you a screenshot, for the moment this is my UV coordinates : 
````kotlin
val vertexUv = ByteBuffer.allocate(6*2*8)
            // It is important to respect the native byte order
            .order(ByteOrder.nativeOrder())
            .putUv(UV(0f,0f))
            .putUv(UV(0f,1f))
            .putUv(UV(1f,1f))
            .putUv(UV(0f,0f))
            .putUv(UV(1f,1f))
            .putUv(UV(1f,0f))
            // Make sure the cursor is pointing in the right place in the byte buffer
            .flip()
````

**Reply by inveders** [reply]
<img width="266" alt="ar" src="https://user-images.githubusercontent.com/40520645/127234530-f42c1c8d-7852-4804-9b8f-da646a5a4ea2.png">
@bejado This is the screenshot
Do not pay attention to planes (grid)

**Reply by bejado** [reply]
The UV coordinates aren't getting set correctly for the right triangle. Can you show more of your code? What does your index buffer look like?

**Reply by inveders** [reply]
@bejado Yes this is my indexData and indexBuffer (Do you need more code?) So What is the good way to set the UV?
````kotlin
// Create the indices
        val indexData = ByteBuffer.allocate(6 * shortSize)
            .order(ByteOrder.nativeOrder())
            .putShort(0)
            .putShort(1)
            .putShort(2)
            .putShort(0)
            .putShort(2)
            .putShort(3)
            .flip()

        indexBuffer = IndexBuffer.Builder()
            .indexCount(6)
            .bufferType(IndexBuffer.Builder.IndexType.USHORT)
            .build(mEngine)

        indexBuffer.setBuffer(mEngine, indexData)
````

**Reply by inveders** [reply]
I found the solution, thanks for the advice. The issue were with the number of points in my vertexUv, I change with 4 points and not 6 :

````kotlin
val vertexUv = ByteBuffer.allocate(6*2*4)
            // It is important to respect the native byte order
            .order(ByteOrder.nativeOrder())
            .putUv(UV(0f,0f))
            .putUv(UV(0f,1f))
            .putUv(UV(1f,1f))
            .putUv(UV(1f,0f))
            // Make sure the cursor is pointing in the right place in the byte buffer
            .flip()
````

Now, all is good !
